metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/extractor/ReferenceSequenceExtractorFromAnalysis.java

```diff
@@ -0,0 +1,85 @@
+/*
+ *
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package uk.ac.ebi.ampt2d.metadata.importer.extractor;
+
+import com.querydsl.core.types.dsl.BooleanExpression;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.QReferenceSequence;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.ReferenceSequence;
+import uk.ac.ebi.ampt2d.metadata.persistence.repositories.ReferenceSequenceRepository;
+import uk.ac.ebi.ena.sra.xml.AnalysisType;
+import uk.ac.ebi.ena.sra.xml.AnalysisType.ANALYSISTYPE.SEQUENCEVARIATION;
+import uk.ac.ebi.ena.sra.xml.ReferenceAssemblyType;
+import uk.ac.ebi.ena.sra.xml.ReferenceSequenceType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ReferenceSequenceExtractorFromAnalysis {
+
+    private static QReferenceSequence qReferenceSequence = QReferenceSequence.referenceSequence;
+
+    private ReferenceSequenceRepository referenceSequenceRepository;
+
+    public ReferenceSequenceExtractorFromAnalysis(ReferenceSequenceRepository referenceSequenceRepository) {
+        this.referenceSequenceRepository = referenceSequenceRepository;
+    }
+
+    public List<ReferenceSequence> getReferenceType(AnalysisType analysisType) {
+        ReferenceSequenceType referenceAlignment = analysisType.getANALYSISTYPE().getREFERENCEALIGNMENT();
+        SEQUENCEVARIATION sequencevariation = analysisType.getANALYSISTYPE().getSEQUENCEVARIATION();
```

Comment:
I believe we should also check `getPROCESSEDREADS` because entries of this type can also have a reference sequence, example: https://www.ebi.ac.uk/ena/data/view/ERZ266973&display=xml

Reply:
Considering that we're going to get `ReferenceSequence`s from its separate query, not from Analysis, it's probably OK to leave it as is for now, seeing as we'll need to delete the ReferenceSequenceExtractorFromAnalysis class after merging #50 anyway. Marking as resolved.